From 7215c43f869d38deb6bef977b7d5d93a1efc2b6b Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:06:36 -0400
Subject: [tools] perf/ui/tui: Add 'F' hotkey to toggle percentage output

Message-id: <1410181752-25631-152-git-send-email-jolsa@redhat.com>
Patchwork-id: 91789
O-Subject: [PATCH RHEL7.1 BZ1134356 151/307] perf ui/tui: Add 'F' hotkey to toggle percentage output
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 105eb30f18197a217695eac4ddf87526f2cba867
Author: Namhyung Kim <namhyung@kernel.org>
Date: Mon Feb 10 11:20:10 2014 +0900

description
===========
Add 'F' hotkey to toggle relative and absolute percentage of filtered
entries.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/ui/browsers/hists.c b/tools/perf/ui/browsers/hists.c
index 7ad1147..4d41698 100644
--- a/tools/perf/ui/browsers/hists.c
+++ b/tools/perf/ui/browsers/hists.c
@@ -1389,6 +1389,7 @@ static int perf_evsel__hists_browse(struct perf_evsel *evsel, int nr_events,
  "C             Collapse all callchains\n"   \
  "d             Zoom into current DSO\n"    \
  "E             Expand all callchains\n"    \
+ "F             Toggle percentage of filtered entries\n"  \
 
  /* help messages are sorted by lexical order of the hotkey */
  const char report_help[] = HIST_BROWSER_HELP_COMMON
@@ -1494,6 +1495,9 @@ static int perf_evsel__hists_browse(struct perf_evsel *evsel, int nr_events,
    if (env->arch)
     tui__header_window(env);
    continue;
+  case 'F':
+   symbol_conf.filter_relative ^= 1;
+   continue;
   case K_F1:
   case 'h':
   case '?':
-- 
1.7.1